corehq/apps/locations/tasks.py

```diff
@@ -100,35 +103,39 @@ def download_locations_async(domain, download_id, include_consumption, headers_o
-    importer = MultiExcelImporter(import_locations_async, file_ref_id)
-    user = CouchUser.get_by_user_id(user_id)
-    results = new_locations_import(domain, importer, user)
-    importer.mark_complete()
-
-    if LOCATIONS_IN_UCR.enabled(domain):
-        # We must rebuild datasources once the location import is complete in
-        # case child locations were not updated, but a parent location was.
-        # For example if a state was updated, the county may reference the state
-        # and need to have its row updated
-        datasources = get_datasources_for_domain(domain, "Location", include_static=True)
-        for datasource in datasources:
-            rebuild_indicators_in_place.delay(
-                datasource.get_id, initiated_by=user.username, source='import_locations'
+    try:
+        importer = MultiExcelImporter(import_locations_async, file_ref_id)
+        user = CouchUser.get_by_user_id(user_id)
+        results = new_locations_import(domain, importer, user)
+        importer.mark_complete()
+
+        if LOCATIONS_IN_UCR.enabled(domain):
+            # We must rebuild datasources once the location import is complete in
+            # case child locations were not updated, but a parent location was.
+            # For example if a state was updated, the county may reference the state
+            # and need to have its row updated
+            datasources = get_datasources_for_domain(domain, "Location", include_static=True)
+            for datasource in datasources:
+                rebuild_indicators_in_place.delay(
+                    datasource.get_id, initiated_by=user.username, source='import_locations'
+                )
+
+        if getattr(settings, 'CELERY_TASK_ALWAYS_EAGER', False):
+            # Log results because they are not sent to the view when
+            # CELERY_TASK_ALWAYS_EAGER is true
+            logging.getLogger(__name__).info(
+                "import_locations_async %s results: %s -> success=%s",
+                file_ref_id,
+                " ".join(
+                    "%s=%r" % (name, getattr(results, name))
+                    for name in ["messages", "warnings", "errors"]
+                    if getattr(results, name)
+                ),
+                results.success,
             )
-
-    if getattr(settings, 'CELERY_TASK_ALWAYS_EAGER', False):
-        # Log results because they are not sent to the view when
-        # CELERY_TASK_ALWAYS_EAGER is true
-        logging.getLogger(__name__).info(
-            "import_locations_async %s results: %s -> success=%s",
-            file_ref_id,
-            " ".join(
-                "%s=%r" % (name, getattr(results, name))
-                for name in ["messages", "warnings", "errors"]
-                if getattr(results, name)
-            ),
-            results.success,
-        )
+    except Exception as e:
+        results = LocationUploadResult()
+        results.errors = [str(e)]
```

Comment:
Can you `notify_exception` here as well?  We'll probably want this in sentry.

Reply:
yikes! swallowed that,
https://github.com/dimagi/commcare-hq/pull/27530